Bottom: d3d1e4b37db293e56e2d8eaff0877e0e0811481e
Top:    7ac97ae731683fb5e2dc2f237e0e6439bf98d538
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 15:47:08 +0100

Remove hack in draw.c which broke PS (and probably other) output

The set_layer call hacked into DrawLayerGroup to reset compositing tricked
the PS exporter to re-open and overwrite output in the main layer output
file during the fab drawing output.


---

diff --git a/src/draw.c b/src/draw.c
index 831a993..0c3b45c 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -770,8 +770,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   int n_entries = PCB->LayerGroups.Number[group];
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index c1124d3..2dfe44b 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -334,10 +334,6 @@ draw_layer (LayerType *layer, void *userdata)
   /* print the non-clearing polys */
   r_search (layer->polygon_tree, drawn_area, NULL, poly_callback, &info);
 
-  /* HACK: Subcomposite polygons separately from other layer primitives */
-  /* Reset the compositing */
-  gui->set_layer (0, group, 0);
-
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
